src/main/java/com/minecolonies/coremod/commands/TreeNode.java

```diff
@@ -0,0 +1,70 @@
+package com.minecolonies.coremod.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// From https://stackoverflow.com/a/40622616/1394993
+
+public class TreeNode<T>
+{
+    private T data = null;
+    private List<TreeNode<T>> children = new ArrayList<>();
+    private TreeNode<T> parent = null;
+
+    public TreeNode(final T data)
+    {
+        this.data = data;
+    }
+
+    public void addChild(final TreeNode<T> child)
+    {
+        child.setParent(this);
+        this.children.add(child);
+    }
+
+    public void addChild(final T childData)
+    {
+        final TreeNode<T> newChild = new TreeNode<>(childData);
+        newChild.setParent(this);
+        children.add(newChild);
+    }
+
+    public void addChildren(final List<TreeNode<T>> moreChildren)
+    {
+        for (final TreeNode<T> t : moreChildren)
+        {
+            t.setParent(this);
+        }
+        this.children.addAll(moreChildren);
+    }
+
+    public List<TreeNode<T>> getChildren()
+    {
+        return children;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "children". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)